onnxruntime/test/providers/cpu/nn/qlinearconv_op_test.cc

```diff
@@ -3,43 +3,101 @@
 
 #include "gtest/gtest.h"
 #include "test/providers/provider_test_utils.h"
-using namespace std;
+
 namespace onnxruntime {
 namespace test {
 
 namespace {
 
-void FindMinMax(const vector<float>& vec, float* min,
-                float* max) {
-  *min = *max = 0;
-  *min = *std::min_element(vec.begin(), vec.end());
-  *max = *std::max_element(vec.begin(), vec.end());
-}
+struct QuantizedTensor {
+  QuantizedTensor(const std::vector<float>& data) {
+    // find input range min and max
+    auto min = *std::min_element(data.begin(), data.end());
+    auto max = *std::max_element(data.begin(), data.end());
+
+    // ensure the data range includes zero
+    min = std::min(min, 0.f);
+    max = std::max(max, 0.f);
+
+    float qmin = std::numeric_limits<uint8_t>::min();
+    float qmax = std::numeric_limits<uint8_t>::max();
+
+    // compute scale and zero point
+    scale_ = (max - min) / (qmax - qmin);
+    const auto initial_zero_point = qmin - min / scale_;
+    zero_point_ = static_cast<uint8_t>(std::round(std::max(qmin, std::min(qmax, initial_zero_point))));
```

Comment:
I am not following this. initial_zero_point would be less than qmin. Then the min of qman and initial_zero_point is selected. Would not the result always be the rounded qmin?